router/throttler/throttler.go

```diff
@@ -0,0 +1,166 @@
+package throttler
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/EagleChen/restrictor"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+var (
+	parameters map[string]map[string]interface{}
+)
+
+//Throttler is an interface for throttling functions
+type Throttler interface {
+	LimitReached(key string) bool
+	Enabled() bool
+}
+
+//HandleT is a Handle for event limiter
+type HandleT struct {
+	UserEventOrderingRequired bool
+	permit                    bool
+	restrictor                restrictor.Restrictor
+	destination               string
+	eventLimit                int
+	timeWindow                time.Duration
+	bucketsInWindow           int
+	userLevelThrottling       bool
+	userLevelThrottler        *UserHandleT
+}
+
+//UserHandleT is a handle for user level event limiter
+type UserHandleT struct {
+	permit          bool
+	restrictor      restrictor.Restrictor
+	destination     string
+	eventLimit      int
+	timeWindow      time.Duration
+	bucketsInWindow int
+}
+
+func init() {
+	loadConfig()
+}
+
+func loadConfig() {
+	parameters = map[string]map[string]interface{}{
+		// https://customer.io/docs/api/#api-documentationlimits
+		"CUSTOMERIO": {
+			"limit":                     30,
+			"timeWindowInS":             1,
+			"userEventOrderingRequired": true,
+			"userLevelThrottling":       false,
+		},
+		// https://help.amplitude.com/hc/en-us/articles/360032842391-HTTP-API-V2#upload-limit
+		"AM": {
+			"limit":                     1000,
+			"timeWindowInS":             1,
+			"userEventOrderingRequired": true,
+			"userLevelThrottling":       true,
+			"userLevelLimit":            10,
+			"userLevelTimeWindowInS":    1,
+		},
+	}
+}
+
+func (throttler *HandleT) setLimits() {
+	// set eventLimit
+	defaultLimit, _ := parameters[throttler.destination]["limit"].(int)
```

Comment:
Should we handle the error for other destinations like GA which don't have throttles?